.github/workflows/iai_benchs_main.yml

```
@@ -0,0 +1,33 @@
+name: CI
+
+on:
+  push:
+    branches: [ main ]
+
+jobs:
+  cache_iai_benchs:
+    name: Cache iai benchs of main
+    runs-on: ubuntu-20.04
+    steps:
+    - name: Install valgrind
+      run: |
+        sudo apt update
+        sudo apt-get install -y valgrind
+    - name: Install stable toolchain
+      uses: dtolnay/rust-toolchain@stable
+      with:
+          toolchain: stable
+    - uses: actions/checkout@v3
+    - name: Initialize iai cache for ${{ github.event.pull_request.base.sha }}
```

This was a mistake on my part in the original pipeline. It worked pretty much by miracle. The correct line is:
```suggestion
    - name: Initialize iai cache for ${{ github.sha }}
```
That's because for push to main there's not actual pull request event, and the valid source of your commit hash is the one from the base context.

---

Oh, it worked because it's just the message lol